src/lib/utils/bswap.h

```diff
@@ -31,7 +31,7 @@ inline u16bit reverse_bytes(u16bit val)
 */
 inline u32bit reverse_bytes(u32bit val)
    {
-#if BOTAN_GCC_VERSION >= 430 && !defined(BOTAN_TARGET_CPU_IS_ARM_FAMILY)
+#if BOTAN_GCC_VERSION >= 430 && !defined(BOTAN_TARGET_ARCH_IS_ARM32)
    /*
```

Comment:
I just verified that at least GCC 4.9 (from Android NDK) emits the REV instruction ("Reverse Bytes reverses the byte order in a register."), which should be optimal. clang 3.5 does the same thing.